src/data/list/basic.lean

```diff
@@ -2749,6 +2749,17 @@ begin
     { simpa [hl] } }
 end
 
+lemma pmap_append {p : ι → Prop} (f : Π (a : ι), p a → α) (l₁ l₂ : list ι)
```

Comment:
Can you add a primed version that has `l₁.pmap f h₁ ++ l₂.pmap f h₂` on the RHS?

Reply:
What is the best way of obtaining `h` from `h₁` and `h₂` in the statement? I tried it first, and it got a bit unwieldy (and then I had problems with the proof...).

Reply:
Perhaps somebody else can add that later?

Reply:
```
lemma pmap_append' {α β : Type*} {p : α → Prop} (f : Π (a : α), p a → β) (l₁ l₂ : list α)
  (h₁ : ∀ a ∈ l₁, p a) (h₂ : ∀ a ∈ l₂, p a) :
  (l₁ ++ l₂).pmap f (λ a ha, (list.mem_append.1 ha).elim (h₁ a) (h₂ a)) =
  l₁.pmap f h₁ ++ l₂.pmap f h₂ :=
pmap_append f l₁ l₂ _
```

Reply:
OK, thanks.